include/neml2/models/crystallography/CrystalGeometry.h

```diff
@@ -27,6 +27,8 @@
 #include "neml2/base/Registry.h"
 #include "neml2/models/Data.h"
 
+#include "neml2/base/TensorName.h"
```

Comment:
I don't think we need this here.

Reply:
It's needed for the implementation of both `CrystalGeometry` and `CubicCrystal`.  This one I was lazy, we could just include it in both cxx files if you prefer.

Reply:
Let's include it in cxx files.